rpc/backend/evm_backend_test.go

```diff
@@ -1,30 +1,380 @@
 
 		if tc.expPass {
-			suite.Require().NoError(err)
+			suite.Require().Nil(err)
 			suite.Require().Equal(tc.expBlockNumber, blockNumber)
 		} else {
-			suite.Require().Error(err)
+			suite.Require().NotNil(err)
+		}
+	}
+}
+
+func (suite *BackendTestSuite) TestGetTendermintBlockByNumber() {
+	var block tmtypes.Block
+
+	testCases := []struct {
+		mame                string
+		blocknumber         ethrpc.BlockNumber
+		registerMockQueries func(ethrpc.BlockNumber)
+		found               bool
+		expPass             bool
+	}{
+		{
+			"fail - client error",
+			ethrpc.BlockNumber(1),
+			func(blockNum ethrpc.BlockNumber) {
+				// Client.Block
+				height := blockNum.Int64()
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlockQueriesError(client, height)
+			},
+			false,
+			false,
+		},
+		{
+			"noop - block not found",
+			ethrpc.BlockNumber(1),
+			func(blockNum ethrpc.BlockNumber) {
+				// Client.Block
+				height := blockNum.Int64()
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlockQueriesNotFound(client, height)
+			},
+			false,
+			true,
+		},
+		{
+			"fail - blockNum < 0 with app state height error",
+			ethrpc.BlockNumber(-1),
+			func(_ ethrpc.BlockNumber) {
+				// QueryClient.Params
+				appHeight := int64(1)
+				var header metadata.MD
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterParamsQueriesError(queryClient, &header, appHeight)
+			},
+			false,
+			false,
+		},
+		{
+			"pass - blockNum < 0 with app state height >= 1",
+			ethrpc.BlockNumber(-1),
+			func(blockNum ethrpc.BlockNumber) {
+				// QueryClient.Params
+				appHeight := int64(1)
+				var header metadata.MD
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterParamsQueries(queryClient, &header, appHeight)
+
+				// Client.Block
+				tmHeight := appHeight
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlockQueries(client, tmHeight)
+
+				block = tmtypes.Block{Header: tmtypes.Header{Height: tmHeight}}
+			},
+			true,
+			true,
+		},
+		{
+			"pass - blockNum = 0 (defaults to blockNum = 1 due to a difference between tendermint heights and geth heights",
+			ethrpc.BlockNumber(0),
+			func(blockNum ethrpc.BlockNumber) {
+				// Client.Block
+				height := blockNum.Int64()
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlockQueries(client, height)
+
+				block = tmtypes.Block{Header: tmtypes.Header{Height: height}}
+			},
+			true,
+			true,
+		},
+		{
+			"pass - blockNum = 1",
+			ethrpc.BlockNumber(1),
+			func(blockNum ethrpc.BlockNumber) {
+				// Client.Block
+				height := blockNum.Int64()
+				client := suite.backend.clientCtx.Client.(*mocks.Client)
+				RegisterBlockQueries(client, height)
+
+				block = tmtypes.Block{Header: tmtypes.Header{Height: height}}
+			},
+			true,
+			true,
+		},
+		// TODO why does the "x-cosmos-block-height" always have to be  "1"?
```

Comment:
This seems to use the value defined in `suite.SetupTest()` --> if you set that to `suite.backend.ctx = rpc.ContextWithHeight(5)` this passes and the other calls don't work.

I couldn't figure out why this is the case yet, but maybe this helps

Reply:
removed this for now and will adrress in a a future PR